x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameIndexer.java

```diff
@@ -167,7 +169,20 @@ protected void onFinish(ActionListener<Void> listener) {
 
     @Override
     protected IterationResult<DataFrameIndexerPosition> doProcess(SearchResponse searchResponse) {
-        final CompositeAggregation agg = searchResponse.getAggregations().get(COMPOSITE_AGGREGATION_NAME);
+        final Aggregations aggregations = searchResponse.getAggregations();
+        // Treat this as a "we reached the end".
+        // This should only happen when all underlying indices have gone away. Consequently, there is no more data to read.
+        if (aggregations == null) {
+            logger.info("[" + getJobId() + "] unexpected null aggregations in search response. " +
+                "Concrete indices may have disappeared or have been closed");
+            auditor.info(getJobId(),
+                "Data frame transform unexpectedly stopped seeing data. " +
+                    "Please verify that these indices exist and are open [" +
+                    Strings.arrayToCommaDelimitedString(getConfig().getSource().getIndex()) +
+                    "].");
```

Comment:
I waffled a bit on how to word this. From my understanding, `aggregations` should only ever be null here iff the underlying indices are all closed/deleted

Reply:
yes, it's only `null` if source became empty.

With the given wording, it sounds more like a warning. So I would either change the log/audit level or the wording.